spec/support/thread_helpers.rb

```diff
@@ -0,0 +1,21 @@
+module ThreadHelpers
+  module_function
+
+  # Isolate created crated in a block in their own `ThreadGroup`.
```

Comment:
I recommend a high level read about ThreadGroups, if not familiar: https://ruby-doc.org/core-2.5.0/ThreadGroup.html